sklearn/model_selection/tests/test_split.py

```diff
@@ -2066,3 +2066,87 @@ def test_stratified_splitter_without_y(cv):
     msg = "missing 1 required positional argument: 'y'"
     with pytest.raises(TypeError, match=msg):
         cv.split(X)
+
+
+@pytest.mark.parametrize(
+    "folds,n_groups,group_size",
+    [
+        (333, 333, 1),
+        (2, 41, 5),
+        (3, 5, 16),
+    ],
+)
+def test_group_kfold_is_stable_with_ties(folds, n_groups, group_size):
+    """Verify groups are assigned based on a stable sort, with ties."""
+    groups = np.repeat(np.arange(n_groups), group_size)
+    X = np.arange(n_groups * group_size)
+    gkf = GroupKFold(folds)
+
+    split = [a for a in gkf.split(X, groups=groups)]
+    for i, (_, test) in enumerate(split):
+        # GroupKFold reverses sort, expect groups dealt in reverse
+        expected = X[groups[::-1] % folds == i]
+        assert_array_equal(test, expected)
+
+
+def test_group_kfold_with_no_ties():
```

Comment:
This test does not fail on `main` which I guess is expected if there are no ties, it does not matter whether the sort is stable or not.

Can you explain a bit what this additional test brings, for example maybe it covers a blind spot of the existing tests?

Reply:
I believe it covers a blindspot, about consistency of the assignment order. It is designed to catch any implementation changes to GroupKFold that would break the consistency of the splits between versions.

However, if sklearn doesn't want to guarantee consistency of the group assignments between version s then this test would be unnecessary.